widget/splitcontainer.go

```diff
@@ -0,0 +1,272 @@
+}
+
+func (r *dividerRenderer) MinSize() fyne.Size {
+	if r.divider.split.Horizontal {
+		return fyne.NewSize(getDividerThickness(), getDividerLength())
+	}
+	return fyne.NewSize(getDividerLength(), getDividerThickness())
+}
+
+func (r *dividerRenderer) Refresh() {
+	r.Layout(r.divider.Size())
+}
+
+func (r *dividerRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *dividerRenderer) BackgroundColor() color.Color {
+	if r.divider.hovered {
+		return theme.HoverColor()
+	}
+	return theme.ButtonColor()
+}
+
+func (r *dividerRenderer) Destroy() {
+}
+
+type splitContainerRenderer struct {
+	split   *SplitContainer
+	divider *divider
+	objects []fyne.CanvasObject
+}
+
+func (r *splitContainerRenderer) Layout(size fyne.Size) {
+	var dividerPos, childAPos, childBPos fyne.Position
+	var dividerSize, childASize, childBSize fyne.Size
+	if r.split.Horizontal {
+		x := 0
+		half := (size.Width - getDividerThickness()) / 2
+		childAPos.X = x
+		childASize.Width = half + r.split.offset
+		childASize.Height = size.Height
+		x += childASize.Width
+		dividerPos.X = x
+		dividerSize.Width = getDividerThickness()
+		dividerSize.Height = size.Height
+		x += dividerSize.Width
+		childBPos.X = x
+		childBSize.Width = half - r.split.offset
+		childBSize.Height = size.Height
+	} else {
+		y := 0
+		half := (size.Height - getDividerThickness()) / 2
+		childAPos.Y = y
+		childASize.Width = size.Width
+		childASize.Height = half + r.split.offset
+		y += childASize.Height
+		dividerPos.Y = y
+		dividerSize.Width = size.Width
+		dividerSize.Height = getDividerThickness()
+		y += dividerSize.Height
+		childBPos.Y = y
+		childBSize.Width = size.Width
+		childBSize.Height = half - r.split.offset
+	}
+	r.divider.Move(dividerPos)
+	r.divider.Resize(dividerSize)
+	r.split.ChildA.Move(childAPos)
+	r.split.ChildA.Resize(childASize)
+	r.split.ChildB.Move(childBPos)
+	r.split.ChildB.Resize(childBSize)
+	canvas.Refresh(r.divider)
+	canvas.Refresh(r.split.ChildA)
+	canvas.Refresh(r.split.ChildB)
+}
+
+func (r *splitContainerRenderer) MinSize() fyne.Size {
+	s := fyne.NewSize(0, 0)
+	for _, o := range r.objects {
+		min := o.MinSize()
+		if r.split.Horizontal {
+			s.Width += min.Width
+			s.Height = fyne.Max(s.Height, min.Height)
+		} else {
+			s.Width = fyne.Max(s.Width, min.Width)
+			s.Height += min.Height
+		}
+	}
+	return s
+}
+
+func (r *splitContainerRenderer) Refresh() {
+	r.Layout(r.split.Size())
+}
+
+func (r *splitContainerRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *splitContainerRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *splitContainerRenderer) Destroy() {
+}
+
+// Declare conformity with CanvasObject interface
+var _ fyne.CanvasObject = (*SplitContainer)(nil)
+
+// SplitContainer defines a container whose size is split between two children.
+type SplitContainer struct {
+	BaseWidget
+	Horizontal     bool
+	ChildA, ChildB fyne.CanvasObject
```

Comment:
Is there some more intuitive naming that could be used here?

Reply:
Ack